### DISCUSSION TITLE
Does rsbuild support react compiler for react 17 and 18 ?

### DISCUSSION THREAD

**Discussion by Romej** [web-infra-dev/rsbuild#4197] [General]
According to the [docs](https://rsbuild.dev/guide/framework/react#react-compiler), we must upgrade to react 19 to use the react compiler.

Is that correct or can we use the compiler + rsbuild with react 17 and 18 ?

The official docs from react point out that the [compiler supports 17 and 18](https://rsbuild.dev/guide/framework/react#react-compiler) so I'm wondering why rsbuild wouldn't allow it considering it is used by many legacy projects that cannot upgrade their dependencies easily.

**Comment by Romej** [comment]
I would suggest specifying that it can be used with 17 and 18 through the config :

```ts
    pluginBabel({
      include: /\.(?:jsx|tsx)$/,
      babelLoaderOptions(opts) {
        opts.plugins?.unshift(['babel-plugin-react-compiler', { target: '17' }]);
      },
    }),
```

**Comment by chenjiahan** [comment]
When we write the Rsbuild + React Compiler documentation, React Compiler only supports React 19.

Now React Compiler has supported React 17 and 18, I have updated the Rsbuild documentation for React 17 and 18, see: https://github.com/web-infra-dev/rsbuild/pull/4198
